Source/CBLSyncConnection+Pull.m

```diff
@@ -162,11 +164,17 @@ - (void) handleIncomingRevision: (BLIPRequest*)request {
     NSDictionary* attachments = nil;
     NSString* docID;
     NSData* json = request.body;
-    if (memmem(json.bytes, json.length, "\"_attachments\":", 15) != NULL) {
+    if (self.onSyncAccessCheck && memmem(json.bytes, json.length, "\"_attachments\":", 15) != NULL) {
```

Comment:
I intended to use || not &&, that was the bug. I have docID there so that (#726) if we allow users to grant access via the listener delegate, the users can use doc ids for in their logic. We can chat about this next week.